playbooks/common/openshift-cluster/upgrades/docker/upgrade.yml

```diff
@@ -0,0 +1,44 @@
+---
+# We need docker service up to remove all the images, but these services will keep
+# trying to re-start and thus re-pull the images we're trying to delete.
+- name: Stop containerized services
+  service: name={{ item }} state=stopped
+  with_items:
+    - "{{ openshift.common.service_type }}-master"
+    - "{{ openshift.common.service_type }}-master-api"
+    - "{{ openshift.common.service_type }}-master-controllers"
+    - "{{ openshift.common.service_type }}-node"
+    - etcd_container
+    - openvswitch
+  failed_when: false
+  when: openshift.common.is_containerized | bool
```

Comment:
Yeah so with this PR, technically you could upgrade Docker by setting an explicit docker_version in inventory, it just wouldn't be a "safe" upgrade with node evac. That seems pretty tough right now, though I guess openshift_docker *could* handle logic across docker/node roles and determine if an evac is necessary, however if we wait for 1.12 (can restart daemon without losing containers) or system containers, then the problem goes away entirely I think. So I think this would be very possible soon, and I'm pretty intrigued by your idea to push ALL upgrade into config.yml runs. Let the roles handle it just by interpreting your inventory, so upgrade would become a process of just setting inventory var, likely openshift_release.

Should be able to do this but it feels like we should just wait until 1.12 or sys containers.